app/scripts/components/exploration/atoms.ts

```diff
@@ -0,0 +1,37 @@
+import { atom } from 'jotai';
+import {
+  DateRange,
+  HEADER_COLUMN_WIDTH,
+  RIGHT_AXIS_SPACE,
+  TimelineDataset,
+  ZoomTransformPlain
+} from './constants';
+
+// Datasets to show on the timeline and their settings
+export const timelineDatasetsAtom = atom<TimelineDataset[]>([]);
```

Comment:
Have you tried playing with [atomWithHash](https://jotai.org/docs/integrations/location#atomwithhash) ?